lib/exceptions.rb

```diff
@@ -36,4 +36,17 @@ def initialize(host)
       super()
     end
   end
+
+  HTTP_CONNECTION_ERRORS = [
+    Addressable::URI::InvalidURIError,
```

Comment:
This whole PR makes a lot of sense to me and I can see how this will help when migrating http libraries.

But I wonder if it might be doing too much. I am picking this exception here specifically, because I know we use `Adressable::URI` all over the place. But the same may apply to others in this list as well.

As I said, `Adresseable::URI` is something we use a lot, and not just in the context of making HTTP requests. My worry is that including this here may hide that this exception is occuring in a context where we would want to know about it (because it is an actual bug or something we would want to work around). Especially when this is used with `rescue_from` it might be caught in more circumstances than it currently is.

Maybe it would suffice to only group the exceptions that are in the `HTTP` namespace here?

Reply:
Yeah ... "prep for migration" was the motivation, but I also think it's a nice clean up on it's own. I tried to stick to error classes and usages which were pretty specifically in the "a request has an issue" realm -- but totally agreed that there could be specificity or ordering issues hiding in how the rescues are arranged.

I might prepare "only HTTP-descended errors" as separate PR, and leave this open for rebase/followup to see if we can become more confident in wrapping more of them.